### DISCUSSION TITLE
Remove files from Apple Photos app?

### DISCUSSION THREAD

**Discussion by dalisoft** [RhetTbull/osxphotos#787] [Ideas]
Hey @RhetTbull 

Recently by my mistaken i'm imported "Wallpapers" into Photos and found it later. I want remove them from "Photos Library", i can these filenames by listing over original "wallpapers" folders. I'm already manually deleted from exported folder but i don't know how to remove from "Apple Photos Library".

Thank you

**Comment by RhetTbull** [comment]
Hi @dalisoft I do not understand exactly what you are asking. It sounds like you mistakenly imported some photos into Apple Photos and now you want to remove those same folders from the Photos library?  

Do you know the approximate date you imported them to Photos? If so, you can go to `LIbrary > Imports` in Apple Photos then find the photos and delete them from the library.

If you're asking how to do this with osxphotos and for some reason `Library > Imports` doesn't work, you could try saving the following to a shell script and running it, for example `sh wallpaper.sh`.  The script will search for all files in a certain folder with names in Photos that match and put them all into an Album in Photos so you can then do whatever you like with them. Of course, there could be other photos in your library that match the name so be careful to examine them before deleting.

In the script change PHOTO_FOLDER to the path to your wallpaper images and PHOTO_ALBUM to the name of the album you want to use.

```sh
#!/bin/sh

# This script iterates over files in a director and adds photos from the Photos library
# matching those file names to a new album in the Photos library

PHOTO_FOLDER=~/Desktop/export
PHOTO_ALBUM="Test Album"

query_command=(osxphotos query)
for f in $PHOTO_FOLDER/*; do
	filename=$(basename "$f")
	query_command+=(--name "\"${filename}\"")
done

query_command+=(--add-to-album "\"${PHOTO_ALBUM}\"")

echo "Running command: ${query_command[@]}"

# run the command
eval "${query_command[@]}"
```
